bundles/io/org.openhab.io.transport.mqtt/src/main/java/org/openhab/io/transport/mqtt/MqttService.java

```diff
@@ -91,7 +91,13 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
             } else if (property.equals("async")) {
                 conn.setAsync(Boolean.parseBoolean(value));
             } else if (property.equals("clientId")) {
+                if (value.length() > 65535) {
+                    logger.warn("clientId must be less than 65536 characters long");
+                    throw new ConfigurationException("clientId", "clientId must be less than 65536 characters long");
+                }
                 conn.setClientId(value);
+            } else if (property.equals("allowLongerClientIds")) {
+                conn.setAllowLongerClientIds(Boolean.parseBoolean(value));
```

Comment:
I'm saying that I read the code that 

```
allowLongerClientIds=true
clientId=0123456789012345678901234
```

will behave differently from

```
clientId=0123456789012345678901234
allowLongerClientIds=true
```

which should not be -- both should act in the same way, doing what the user intends.